Installation using npm fails for latest version (0.1.3)

When trying to install `tree-sitter-swift`, I get the following error:

```
root@d0f60f746995:/# npm install -g tree-sitter-swift
npm notice 
npm notice New minor version of npm available! 8.3.1 -> 8.5.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.5.1
npm notice Run npm install -g npm@8.5.1 to update!
npm notice 
npm ERR! code 1
npm ERR! path /usr/lib/node_modules/tree-sitter-swift
npm ERR! command failed
npm ERR! command sh -c tree-sitter generate
npm ERR! node:events:498
npm ERR!       throw er; // Unhandled 'error' event
npm ERR!       ^
npm ERR! 
npm ERR! Error: spawn /usr/lib/node_modules/tree-sitter-swift/node_modules/tree-sitter-cli/tree-sitter ENOENT
npm ERR!     at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
npm ERR!     at onErrorNT (node:internal/child_process:478:16)
npm ERR!     at processTicksAndRejections (node:internal/process/task_queues:83:21)
npm ERR! Emitted 'error' event on ChildProcess instance at:
npm ERR!     at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
npm ERR!     at onErrorNT (node:internal/child_process:478:16)
npm ERR!     at processTicksAndRejections (node:internal/process/task_queues:83:21) {
npm ERR!   errno: -2,
npm ERR!   code: 'ENOENT',
npm ERR!   syscall: 'spawn /usr/lib/node_modules/tree-sitter-swift/node_modules/tree-sitter-cli/tree-sitter',
npm ERR!   path: '/usr/lib/node_modules/tree-sitter-swift/node_modules/tree-sitter-cli/tree-sitter',
npm ERR!   spawnargs: [ 'generate' ]
npm ERR! }
```
Even after installing `tree-sitter-cli`, this error shows up. I can install version (0.1.1) without problems.

You can replicate this by running an Ubuntu docker container with `docker run -it ubuntu bash` and executing the following commands in it:
```
apt update && apt install python3 build-essential curl
curl -fsSL https://deb.nodesource.com/setup_16.x | bash -
apt install nodejs
npm install -g tree-sitter-swift
```

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -34,4 +34,5 @@ jobs:
       env:
           PREFIX: /tmp
     - run: cargo test
-    - run: cd ./test-npm-package && npm test; cd ..
+    - run: cd ./test-npm-package && npm test; cd -
+    - run: npm pack && cd .. && npm install -g ./tree-sitter-swift/tree-sitter-swift-*.tgz; cd -

--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "description": "A tree-sitter grammar for the Swift programming language.",
   "main": "bindings/node/index.js",
   "scripts": {
-    "install": "tree-sitter generate",
+    "install": "node scripts/wait-for-tree-sitter.js && tree-sitter generate",
     "postinstall": "node-gyp configure && node-gyp build",
     "ci": "prettier --check grammar.js",
     "test-ci": "./scripts/test-with-memcheck.sh --install-valgrind",
@@ -25,9 +25,9 @@
   },
   "homepage": "https://github.com/alex-pinkus/tree-sitter-swift#readme",
   "dependencies": {
-    "@types/node": "^16.11.10",
     "nan": "^2.15.0",
-    "tree-sitter-cli": "^0.20.0"
+    "tree-sitter-cli": "^0.20.0",
+    "which": "2.0.2"
   },
   "devDependencies": {
     "node-gyp": "^8.4.1",

--- a/scripts/wait-for-tree-sitter.js
+++ b/scripts/wait-for-tree-sitter.js
@@ -1,0 +1,77 @@
+const fs = require("fs");
+const os = require("os");
+const path = require("path");
+const which = require("which");
+const { promisify } = require("util");
+const stat = promisify(fs.stat);
+
+async function main() {
+  const treeSitterExecutable = await which("tree-sitter");
+  if (!treeSitterExecutable.includes("node_modules")) {
+    // Not installed through npm, so should be safe.
+    return;
+  }
+
+  const realTreeSitterDir = path.join(
+    treeSitterExecutable,
+    "..",
+    "..",
+    "tree-sitter-cli"
+  );
+  let timeout = undefined;
+  let timeoutResolve = undefined;
+  Promise.race([
+    waitForOneOf(realTreeSitterDir, ["tree-sitter", "tree-sitter.exe"]).then(
+      () => {
+        clearTimeout(timeout);
+        timeoutResolve();
+      }
+    ),
+    new Promise((resolve) => {
+      timeoutResolve = resolve;
+      timeout = setTimeout(resolve, 10000);
+    }),
+  ]);
+}
+
+async function waitForOneOf(dir, files) {
+  for (const file of files) {
+    try {
+      if (await canExecute(path.join(dir, file))) {
+        return;
+      }
+    } catch {
+      // File doesn't yet exist -- we must wait for it.
+    }
+  }
+
+  await new Promise((resolve) => {
+    try {
+      fs.watch(dir, { persistent: false }, (eventType, filename) => {
+        if (
+          (eventType !== "rename" || os.platform() !== "win32") &&
+          files.includes(filename)
+        ) {
+          let resolved = false;
+          canExecute(path.join(dir, filename)).then((canExec) => {
+            if (canExec && !resolved) {
+              resolve();
+              resolved = true;
+            }
+          });
+        }
+      });
+    } catch (err) {
+      if (err.name !== "AbortError") {
+        console.error(err);
+      }
+    }
+  });
+}
+
+async function canExecute(filePath) {
+  const fileStat = await stat(filePath);
+  return fileStat.mode & 0111 || os.platform() === "win32";
+}
+
+main();

